src/d3a/models/read_user_profile.py

```diff
@@ -322,3 +322,9 @@ def read_arbitrary_profile(profile_type: InputProfileTypes,
             return _calculate_energy_from_power_profile(filled_profile, GlobalConfig.slot_length)
         else:
             return filled_profile
+
+
+def read_and_convert_identity_profile_to_float(profile):
```

Comment:
Can you please tell me the reason for this additional function? Shouldn't `read_arbitrary_profile deal` (or rather its child _read_from_different_sources_todict ) deal with the conversion from json to float dict?

Reply:
Yes, but it did it only for the SimulationConfig class, and specifically for its market_maker_rate member. I needed it though as a free function, in order to apply the same conversion on the market maker rate on the constructor of the MarketMakerStrategy class, and I did not want to duplicate the code. Hence, this calculation had to be refactored to a free function.

Reply:
Ok, thanks! Makes sense.